server/views/pages/appointments/search/results.njk

```diff
@@ -172,23 +172,22 @@
             },
             clearLink: {
                 text: 'Clear filters',
-                href: '?startDate=' + startDate | toDate | toDateString
+                href: '?startDate=' + startDate
             },
             categories: selectedFilterCategories
         } %}
     {% endif %}
 
     {% set filterOptionsHtml %}
-        {{ datePicker({
+        {{ hmppsDatepicker({
             id: 'startDate',
             name: 'startDate',
             label: {
                 text: "Date",
                 classes: "govuk-label--s"
             },
-            renderedErrorMessage: validationErrors | findError('startDate'),
-            validationErrors: validationErrors,
-            formResponses: formResponses.startDate or startDate
+            errorMessage: validationErrors | findError('startDate'),
+            value: (formResponses.startDate or startDate) | isoDateToDatePickerDate
```

Comment:
Keep the date as an ISO string and only convert back to the date picker specific format when displaying